src/server/engine_shard_set.h

```diff
@@ -153,6 +153,16 @@ class EngineShard {
 
   void CacheStats();
 
+  // We are running a task that checks whether we need to
+  // do memory de-fragmentation here, this task only run
+  // when there are available CPU time.
+  void StartDefragTask(util::ProactorBase* pb);
+
+  // scan the shard with the cursor and apply
+  // de-fragmentation option for entries. This function will return the new cursor at the end of the
+  // scan This function is called from context of StartDefragTask
+  uint64_t DoDefragIfRequired(uint64_t cursor);
```

Comment:
this function always scans and defrags at the table level. why "IfRequired"  suffix?

Reply:
I wasn't sure if to add this if, so if you don't see the need I will revert to the simpler name (which I personally preferred anyway)